### PR TITLE
feat: Add GetAllClientIDsOfType query function 

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #2844 

### Commit Message / Changelog Entry

```bash
feat: add `GetAllClientIDsOfType` to 02-client keeper.
feat: add `PrefixedClientStorePath` and `PrefixedClientStoreKey` functions to 24-host
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ### Features
 
+* (core/02-client) [\#2855](https://github.com/cosmos/ibc-go/pull/2855) Add `GetAllClientIDsOfType` to 02-client keeper.
+* (core/24-host) [\#2855](https://github.com/cosmos/ibc-go/pull/2855) Add `PrefixedClientStorePath` and `PrefixedClientStoreKey` functions to 24-host. Constructs a prefixed key path using the client type.
```
Comment (colin-axner) [inline CHANGELOG.md:119]:
```suggestion
* (core/24-host) [\#2855](https://github.com/cosmos/ibc-go/pull/2855) Add `PrefixedClientStorePath` and `PrefixedClientStoreKey` functions to 24-host
```

File: modules/core/02-client/keeper/keeper.go [hunk 2]
```diff
@@ -383,6 +383,28 @@ func (k Keeper) GetAllClients(ctx sdk.Context) (states []exported.ClientState) {
 	return states
 }
 
+// GetAllClientIDsOfType will iterate over the provided client type prefix in the client store
+// and return a list of clientIDs associated with the client type.
+func (k Keeper) GetAllClientIDsOfType(ctx sdk.Context, clientType string) (clientIDs []string) {
```
Comment (colin-axner) [inline modules/core/02-client/keeper/keeper.go:388]: This will be used in a followup pr

**Review by charleenfei — APPROVED** [review]
nice addition!

File: modules/core/02-client/keeper/keeper.go (hunk 2 above)
Comment (chatton) [reply modules/core/02-client/keeper/keeper.go:388]: I have a preference for not using named return values e.g. `(clientIDs []string)`. I think it's more readable to be more explicit with and just declare the slice next to its usage

i.e.

```golang
        var clientIDs []string
	for ; iterator.Valid(); iterator.Next() {
                ...
		clientIDs = append(clientIDs, clientID)
	}
```

WDYT?

**Comment by colin-axner** [comment]
@chatton and I realized it's possible to achieve the desired functionality by adding a prefix into the `IterateClients` function so we will go that direction. I will open a followup pr

**Pull request closed without merging**